pkg/broker/broker.go

```diff
@@ -339,12 +339,38 @@ func (a AnsibleBroker) Bind(instanceUUID uuid.UUID, bindingUUID uuid.UUID, req *
 		Parameters: &params,
 	}
 
+	// Verify we're not rebinding the same instance. if err is nil, there is an
+	// instance. Let's compare it to the instance we're being asked to bind.
+	//
+	// if err is not nil, we will just bubble that up
+	//
 	// if binding instance exists, and the parameters are the same return: 200.
 	// if binding instance exists, and the parameters are different return: 409.
 	//
 	// return 201 when we're done.
-	//
-	// once we create the binding instance, we call apb.Bind
+	if bi, err := a.dao.GetBindInstance(bindingUUID.String()); err == nil {
+		if bi.Id.String() == bindingInstance.Id.String() &&
```

Comment:
I think there's a slight improvement you can make here. What if you check if the uuid matches first, then inside that if block do a single if block ```if reflect.DeepEqual(bi.Parameters, bindingInstance.Parameters) { ... } else { ... }```.  If you do it that way you will only have to run the DeepEqual function once instead of twice. That should make this slightly more efficient since DeepEqual does a bunch of loops to compare values on the provided data structure.